### PR TITLE
check for valid stepper driver types

### PR DESCRIPTION
**Pull request by ellensp**
### Requirements

#define a stepper driver type

### Description

Many people cant seem to type the stepper driver types correctly. This checks defined stepper types are valid.
 
### Benefits

Stops typos in stepper driver types.

### Related Issues

Hoping to stop issues like https://reprap.org/forum/read.php?1,878052

### CODE REVIEW

**Comment by ellensp** [comment]
If you can tidy all the #ifdefs  in SanityCheck.h  with a clever macro, please assist.

File: Marlin/src/inc/SanityCheck.h [hunk 1]
```diff
@@ -3173,5 +3173,89 @@ static_assert(   _ARR_TEST(3,0) && _ARR_TEST(3,1) && _ARR_TEST(3,2)
   #endif
 #endif
 
+/**
+ * Sanity check for valid stepper driver types
+ */
+#ifdef X_DRIVER_TYPE
```
Comment (sjasonsmith) [inline Marlin/src/inc/SanityCheck.h]: If you redefine “known” to include unspecified, then you could move these defines checks into the shared macro.

**Comment by sjasonsmith** [comment]
I made this mistake myself a couple days ago. The sanity check would be nice. In my case I set one driver to “TMC_228”.

**Comment by ellensp** [comment]
I like this, it will cause some rumblings at now having to always define a stepper driver. but I'm ok with that. 
Will need all examples updated with stepper driver types...

**Comment by sjasonsmith** [comment]
> I like this, it will cause some rumblings at now having to always define a stepper driver. but I'm ok with that.
> Will need all examples updated with stepper driver types...

This will resolve on issue, which is that the "default" driver was not exactly the same as a 4988, even though everyone assumes it is. The timings used on the default driver were actually different than if you explicitly declared 4988.

**Comment by thisiskeithb** [comment]
> This will resolve on issue, which is that the "default" driver was not exactly the same as a 4988, even though everyone assumes it is.

Only because of:

https://github.com/MarlinFirmware/Marlin/blob/8285953c1b5be6ecc7ff9103f287776bede8cfb2/Marlin/Configuration.h#L671

I never actually confirmed that was the case though.

**Comment by thinkyhead** [comment]
> I never actually confirmed that was the case though.

It was the case when it was written, but some things were changed to affect the fallback behavior. It will be true once more after this is merged.

**Comment by ellensp** [comment]
Why `#ifndef X_DRIVER_TYPE` vs `#if !defined(Y_DRIVER_TYPE)`  in Conditionals_LCD.h ?

**Comment by thisiskeithb** [comment]
This PR broke multiple TMC driver flags: https://github.com/MarlinFirmware/Marlin/commit/ea80b2c8fc4209160eed377437610c298312e97f#comments

**Comment by ellensp** [comment]
A fix, can some else test https://github.com/ellensp/Marlin/commit/7a266dac0215d60059b74923a4f5968a1c9ee027

**Pull request merged**